### ISSUE TITLE
Not functional with HomeAssistent >= 2022.7

### ISSUE THREAD

**Issue by Tinuz97** [pilotak/homeassistant-clientraw#17]
Since update to the latest version of Home Assistent (
-2022.7.0) the addon does not work (for me) anymore,here is a log :+1: 

Logger: homeassistant
Source: custom_components/clientraw/sensor.py:541
Integration: clientraw (documentation, issues)
First occurred: 6 juli 2022 om 23:15:00 (1 occurrences)
Last logged: 6 juli 2022 om 23:15:00

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/clientraw/sensor.py", line 541, in async_update
    await asyncio.wait(tasks, loop=self.hass.loop)
TypeError: wait() got an unexpected keyword argument 'loop'

**Comment by ktn7419** [comment]
Same thing with me.

**Comment by vk2him** [comment]
I recall you made some changes a few months ago as the loop function had an issue. It seems the Loop argument is no longer supported with 2022.7.0 - refer to this issue logged  a few days ago.

https://github.com/home-assistant/core/issues/74382

I also found mention of this in the 2022.7.0 release blog - I hope you can get it working again as I rely on this a lot - many thanks!
```
raise TypeError(
TypeError: As of 3.10, the *loop* parameter was removed from Lock() since it is no longer necessary
```

**Comment by Kvarnbryggan** [comment]
I have the same problem...

**Comment by pilotak** [comment]
why they always have to break something 😞 i will look into it

**Comment by JochenKlenk** [comment]
same issue here

**Comment by roosenbrug** [comment]
Same problem here.

**Comment by ktn7419** [comment]
Not sure if I found a fix or work-around, but I made this change and has been working for me the past 24 hours.

In sensor.py line 541,  await asyncio.wait(tasks, loop=self.hass.loop)
I changed it to await asyncio.wait(tasks)

restart and then all my clientraw data now appears and updates. No errors or warnings

**Comment by vk2him** [comment]
Thanks @ktn7419  - this worked for me as well - happy days!

**Comment by vk2him** [comment]
I spoke too soon - it seems to be updating once per minute - it used to update every 15 seconds or so before - are you seeing this as well?

**Comment by ktn7419** [comment]
Hi Ian, I did notice that, but I didn't mind that. I just wanted the data. Like I said, workaround,  probably not a permanent fix.
At least data is present now
Regards, Gary

⁣Get BlueMail for Android ​

On Jul 8, 2022, 8:50 AM, at 8:50 AM, Ian Mills ***@***.***> wrote:
>I spoke too soon - it seems to be updating once per minute - it used to
>update every 15 seconds or so before - are you seeing this as well?
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/pilotak/homeassistant-clientraw/issues/17#issuecomment-1178953162
>You are receiving this because you were mentioned.
>
>Message ID:
***@***.***>

**Comment by ktn7419** [comment]
@vk2him  I arrived home and thought of your reply most of the day. Not sure how you can get a 15 sec update when the minimum polling interval is 1 minute. I looked up mine and it is set at a 10 minute interval, so I actually did NOT see an update issue. My values still good.

**Comment by vk2him** [comment]
@ktn7419 - isn't memory a strange thing lol ... yes you're absolutely correct - I have my interval set to one minute so it's working as expected. In my defence, I'm 62 years old lol.
As an aside, before you posted this fix I managed to obtain the clientraw data via mqtt as Weatherdisplay has that ability, so I am getting the mqtt clientraw updates every 15 seconds. I use the clientraw data to drive a live view weather card that I've modified that aggregates data from a number of weather sources. I can now use this card with this clientraw integration or the mqtt method.

![Screen Shot 353](https://user-images.githubusercontent.com/22878629/178080576-c51c01dc-6943-4aa1-ac9a-51e486ef9ce6.jpg)

**Comment by vk6zgd** [comment]
This fix worked for me also , updating every minute.

**Comment by pilotak** [comment]
@ktn7419 good catch, do you want to submit a PR?

**Comment by ktn7419** [comment]
@pilotak I'm new at Github workings. I'll give it a try. ---Looked at the documentation and I'm not seeing "Create a branch" under "Development".

**Comment by pilotak** [comment]
You don't need to create a new branch - just "fork" the repo -> open the file, ammend, save -> click on create pull request -> done. I will handle the rest

**Comment by ktn7419** [comment]
Hello Pavel,

Saturday, July 9, 2022, 8:09:13 AM, you wrote:


You don't need to create a new branch - just "fork" the repo -> open the file, ammend, save -> click on create pull request -> done. I will handle the rest
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Thanks for info. Done.

-- 
Best regards,
 Gary                            ***@***.***

**Comment by pilotak** [comment]
fixed #18
